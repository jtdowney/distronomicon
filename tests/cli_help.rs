use assert_cmd::Command;

#[test]
fn test_help_output() {
    let mut cmd = Command::cargo_bin("distronomicon").unwrap();
    let output = cmd.arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    insta::assert_snapshot!(stdout);
}

#[test]
fn test_check_subcommand_help() {
    let mut cmd = Command::cargo_bin("distronomicon").unwrap();
    let output = cmd.arg("check").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    insta::assert_snapshot!(stdout);
}

#[test]
fn test_update_subcommand_help() {
    let mut cmd = Command::cargo_bin("distronomicon").unwrap();
    let output = cmd.arg("update").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    insta::assert_snapshot!(stdout);
}

#[test]
fn test_version_subcommand_help() {
    let mut cmd = Command::cargo_bin("distronomicon").unwrap();
    let output = cmd.arg("version").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    insta::assert_snapshot!(stdout);
}
