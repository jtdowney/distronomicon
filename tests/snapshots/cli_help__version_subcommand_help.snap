---
source: tests/cli_help.rs
expression: stdout
---
Show currently installed version

Usage: distronomicon --repo <REPO> --app <APP> --pattern <PATTERN> --state-directory <STATE_DIRECTORY> version

Options:
  -h, --help  Print help
